projects/dynamic_sparse/runs/resnet_cifar.py

```diff
@@ -0,0 +1,73 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+from ray import tune
+
+from nupic.research.frameworks.dynamic_sparse.common.utils import run_ray
+
+# experiment configurations
+base_exp_config = dict(
+    device="cuda",
+    dataset_name="CIFAR10",
+    model=tune.grid_search(["BaseModel", "SparseModel"]),
+    data_dir="~/nta/datasets",
+    augment_images=True,
+    epochs=200,
+    # train_batches_per_epoch=400,
+    # ---- network related
+    # network="resnet152",
+    # network="WideResNet",
+    network=tune.grid_search(["resnet152", "WideResNet"]),
+    percent_on_k_winner=tune.grid_search([0.25, 1]),
+    boost_strength=1.5,
+    boost_strength_factor=0.85,
+    k_inference_factor=1.0,
+    sparse_type="precise_per_output",
+    sparse_start=1,
+    sparse_end=None,
+    on_perc=0.5,
+    dropout_rate=0,  # zero dropout in wideresnet
+    # ---- optimizer related
+    optim_alg="SGD",
+    learning_rate=0.1,
+    lr_scheduler="MultiStepLR",
+    # lr_milestones=[60, 90, 110],
+    lr_milestones=[60, 120, 160],
+    lr_gamma=0.2,
+    weight_decay=0.0005,
+    momentum=0.9,
```

Comment:
BTW, I have quite often found that momentum hurts sparse networks. It makes sense - I can explain it in person. We should use 0 momentum for sparse nets.

Reply:
Thanks. I did see that happen in GSC, but I thought it was a particularity of GSC networks and not of sparse networks. I can see how momentum wouldn't make a lot of sense in dynamic sparse (connection) networks, but I don't see that same case in the static sparse (connection) networks.